ping.go

```diff
@@ -498,19 +498,42 @@ func (p *Pinger) Statistics() *Statistics {
 	return &s
 }
 
+type expBackoff struct {
+	baseDelay time.Duration
+	maxExp    int64
+	c         int64
+}
+
+func (b *expBackoff) Get() time.Duration {
+	if b.c < b.maxExp {
+		b.c++
+	}
+
+	return b.baseDelay * time.Duration(rand.Int63n(1<<b.c))
+}
+
+func newExpBackoff(baseDelay time.Duration, maxExp int64) expBackoff {
+	return expBackoff{baseDelay: baseDelay, maxExp: maxExp}
+}
+
 func (p *Pinger) recvICMP(
 	conn *icmp.PacketConn,
 	recv chan<- *packet,
 	wg *sync.WaitGroup,
 ) error {
 	defer wg.Done()
+
+	// Start by waiting for 100 µs and increase to a possible maximum of ~ 100 ms.
```

Comment:
Pings over virtual interfaces, say in a local VM network, can be sub 100µs. Maybe start this at 50µs? Not sure what you were basing this value on.

Reply:
sounds, good, changed.